lib/internal/fs/promises.js

```diff
@@ -510,18 +510,15 @@ async function open(path, flags, mode) {
 async function read(handle, bufferOrOptions, offset, length, position) {
   let buffer = bufferOrOptions;
   if (!isArrayBufferView(buffer)) {
-    if (bufferOrOptions === undefined) {
-      bufferOrOptions = {};
-    }
-    if (bufferOrOptions.buffer) {
-      buffer = bufferOrOptions.buffer;
-      validateBuffer(buffer);
-    } else {
-      buffer = Buffer.alloc(16384);
-    }
-    offset = bufferOrOptions.offset || 0;
-    length = bufferOrOptions.length ?? buffer.byteLength;
-    position = bufferOrOptions.position ?? null;
+    bufferOrOptions ??= {};
+    ({
+      buffer = Buffer.alloc(16384),
+      offset = 0,
+      length = buffer.byteLength - offset,
+      position = null
+    } = bufferOrOptions);
```

Comment:
```suggestion
    } = bufferOrOptions ?? ObjectCreate(null));
```

Reply:
IMHO destructuring in this case is better. In particular, it might enforce consistency: when different params are defaulted via `??`, `||`, `if (x === undefined)`, `if (x == null)` etc., it's easier to make a mistake.

Totally agreed on null prototype; I'll replace `{}`s in cousin methods as well.

Also on global scale, if creating objects in runtime is the main concern, how about making a single shared object for all those destructurings in core? Like:
```js
const defaultArgs = Object.freeze(Object.create(null));
```

Reply:
> it's easier to make a mistake.

Hopefully we would have enough testing to catch those mistakes, but point taken. Also I didn't read the previous code very carefully, I didn't notice we were already assigning a throw away object in the previous version, oh well.


> Also on global scale, if creating objects in runtime is the main concern, how about making a single shared object for all those destructurings in core?

I wouldn't say it's a main concern, it's more a nice to have optimisation if the same code can run with fewer objects. Having a single shared object could work, like we already have a shared `kEnumerableProperty` object which follows the same logic, that can be a follow up PR.

Reply:
Hint taken as well, added a bit more tests.
I guess it's okay to keep `ObjectCreate(null)` version at this point.

I think improving destructuring with shared object and hoping for some "optional destructuring" syntax in ES202X is the best option.
Or if performance loss is really high and there are benchmarks showing that, maybe it would be a good idea to consolidate such tips in some sort of guideline for core?